lib/icinga/clusterevents.cpp

```diff
@@ -736,6 +736,30 @@ Value ClusterEvents::ExecuteCommandAPIHandler(const MessageOrigin::Ptr& origin,
 	if (!listener)
 		return Empty;
 
+	if (!origin->IsLocal()) {
+		Endpoint::Ptr endpoint = origin->FromClient->GetEndpoint();
+
+		/* Discard messages from anonymous clients */
+		if (!endpoint) {
+			Log(LogNotice, "ClusterEvents") << "Discarding 'execute command' message from '"
+				<< origin->FromClient->GetIdentity() << "': Invalid endpoint origin (client not allowed).";
+			return Empty;
+		}
+
+		Zone::Ptr localZone = Zone::GetLocalZone();
+		bool fromLocalZone = endpoint->GetZone() == localZone;
+
+		Zone::Ptr parentZone = localZone->GetParent();
+		bool fromParentZone = parentZone && endpoint->GetZone() == parentZone;
+
+		/* Discard message that are neither from our local zone nor from our parent zone */
+		if (!fromLocalZone && !fromParentZone) {
```

Comment:
Interesting.. you've not variablized some values used multiple times, but variablized some values used only one time..

Idea:

1. variablize `endpoint->GetZone()`
2. omit `parentZone && `
3. de-variablize `parentZone`
4. de-variablize `fromLocalZone` and `fromParentZone`

Should stay readable, shouldn’t it? Also `fromParentZone` is computed only when neccessary.

Reply:
On 2.: Is it guaranteed that `endpoint->GetZone() != nullptr`?

Reply:
Sure. Don’t you remember the 🐔-and-🥚 problem with creating a zone/endpoint via API?

Reply:
Actually, I don't, but I can imagine what you're referring to (probably you can't create either using the API because it requires the other?).

Anyways, that sounds like something that might be addressed in the feature so I don't see an issue with having more robust code here. Also, the intention between these variables wasn't code deduplication but naming expressions for documentation purposes (I'd even say that the comment above the if is now fully redundant but it's still there from an earlier version that didn't have these variables).

Reply:
Then leave all ex. 1 as-is and remove the comment. The earlier version doesn’t matter.